src/Psalm/Codebase.php

```diff
@@ -1611,6 +1612,51 @@ public function getCompletionItemsForPartialSymbol(
             );
         }
 
+        $functions = $this->functions->getMatchingFunctionNames($type_string, $offset, $file_path, $this);
+
+        $namespace_map = [];
+        if ($aliases) {
+            $namespace_map += $aliases->uses_flipped;
+            if ($aliases->namespace) {
+                $namespace_map[$aliases->namespace] = '';
+            }
+        }
+
+        // Sort the map by longest first, so we replace most specific
+        // used namespaces first.
+        ksort($namespace_map);
+        $namespace_map = array_reverse($namespace_map);
+
+        foreach ($functions as $function) {
+            // Transform FQFN relative to all uses namespaces
+            $function_name = $function->cased_name;
+            if (!$function_name) {
+                continue;
+            }
+            $in_namespace_map = false;
+            foreach ($namespace_map as $namespace_name => $namespace_alias) {
+                if (strpos(strtolower($function_name), $namespace_name . '\\') === 0) {
```

Comment:
Could move strtolower(..) before the loop

Reply:
You can go one step further: though `$functions` is a `list<string>`, it can instead be an `array<lowercase-string, string>` if `getMatchingFunctionNames` is updated.

Reply:
I had kept it at `list<string>` to reflect the return typeof `ClassLIkes::getMatchingClassLikeNames`.

Reply:
I'll update this though, ATM `getAllStubbedFunctions` `reflection::getFunctions()` and `config::getPredefinedFunctions()` all return `array<string, FunctionLike>` but I think in reality though should all be also `lowercase-string`?